### ISSUE TITLE
csv.rb uses postfix exceptions when converters are specified

### ISSUE THREAD

**Issue by headius** [jruby/jruby#1816]
See code in csv.rb, in CSV#convert_fields and the lambda-based converters around line 947.

Simple benchmark to show the devastating effect:

```
system ~/projects/jruby $ jruby -rbenchmark -rcsv -e 'data = "\"one\",\"two\",\"three\"\n\"1\",\"2\",\"3\"".encode("UTF-16BE") * 10_000; loop { puts Benchmark.measure { CSV.parse(data, converters: :integer) } }'
 19.900000   0.230000  20.130000 ( 12.954000)
 12.390000   0.070000  12.460000 ( 11.322000)
^C
system ~/projects/jruby $ rvm ruby-2.1.1 do ruby -rbenchmark -rcsv -e 'data = "\"one\",\"two\",\"three\"\n\"1\",\"2\",\"3\"".encode("UTF-16BE") * 10_000; loop { puts Benchmark.measure { CSV.parse(data, converters: :integer) } }'
  0.820000   0.000000   0.820000 (  0.828674)
  0.820000   0.000000   0.820000 (  0.828970)
```

Without converters, we perform about like MRI.

**Comment by headius** [comment]
Marking to fix for 1.7.23 (and 9.0.2.0).

**Comment by enebo** [comment]
I am closing this.  We did indirectly address performance issue itself on 9k already without applying the better error  reporting logic which was mentioned above.  I think the train for improving general performance on 1.7 should be spent on 9k at this point.  It's dead...
